client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -267,7 +267,12 @@ private static Request getStyleRequest(String method, GetRequest getRequest) {
     }
 
     static Request sourceExists(GetRequest getRequest) {
-        Request request = new Request(HttpHead.METHOD_NAME, endpoint(getRequest.index(), getRequest.type(), getRequest.id(), "_source"));
+        String endpoint = new EndpointBuilder()
```

Comment:
If the request has a type, we should make sure to use the old endpoint (see `RequestConverters#explain` for example). We should also update `RequestConvertersTests` to check both the typeless + typed versions (as in `testExplain` and `testExplainWithType`).

Reply:
Is that for mixed cluster scenarios? Just so I understand it better for next time.

Reply:
As we said above, GetRequest has always a type, if it isn't provided it is "_doc". Would we always want to use the old deprecated endpoint then?

Reply:
> Is that for mixed cluster scenarios?

Right, this allows us to support a scenario where some nodes are running 6.x, and don't understand the new typeless APIs. Also, our goal is to not make breaking changes to the HLRC behavior in 7.0 -- calling the old typed endpoint when a type is provided in a request allows us to maintain the exact same behavior as in 6.x, where we require a match on the type name.

> Would we always want to use the old deprecated endpoint then?

If HLRC requests are constructed without types (as in `new GetRequest("index", "id")`), then we should use the typeless endpoints so that users can move to the new APIs + avoid deprecation warnings. I think of it as an internal implementation detail that a typeless get request executes by pretending the type is `_doc`. Note that if we had two separate `GetRequest` classes (one for the HLRC, one for the server), then we probably would have implemented this differently. For example in the HLRC `TermVectorsRequest`, we chose to use 'null' for the type in typeless requests, since this is a separate class from the server version.

I hope this makes sense, please feel free to ping me and I can try to explain further.